File: 백준/Bronze/5618. 공약수/공약수.java
import java.util.*;
import java.io.*;

public class Main {
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		
		int n = Integer.parseInt(br.readLine());
		int[] arr = new int[n];
		
		st = new StringTokenizer(br.readLine());
		for(int i=0; i<n; i++) {
			arr[i] = Integer.parseInt(st.nextToken());
		}
		
		Arrays.sort(arr);
		
		for(int i=1; i<=arr[0]; i++) {
			int count = 0;
			
			for(int j=0; j<n; j++) {
				if(arr[j] % i == 0) {
					count++;
				}
			}
			
			if(count == n) {
				System.out.println(i);
			}
		}
	}
}

File: 백준/Bronze/5618. 공약수/공약수.cc
#include <iostream>
#include <algorithm>

using namespace std;

int main(void) {

	int n, a, b, c, temp;
	cin >> n;

	int* arr = new int[n];

	for (int i = 0; i < n; i++) {
		cin >> arr[i];
	}

	sort(arr, arr + n);

	for (int i = 1; i <= arr[0]; i++) {
		int count = 0;

		for (int j = 0; j < n; j++) {
			if (arr[j] % i == 0) {
				count++;
			}
		}

		if (count == n) {
			cout << i << "\n";
		}
	}
}